spi/topology/src/main/java/quarks/topology/spi/graph/ConnectorStream.java

```diff
@@ -96,6 +97,28 @@ protected Graph graph() {
     }
 
     @Override
+    public <E extends Enum<E>> EnumMap<E,TStream<T>> split(Class<E> enumClass, Function<T, E> splitter) {
+
+        E[] es = enumClass.getEnumConstants();
+        if(es == null) {
```

Comment:
I'm merging this pull request but I'm not sure I understand this code:

E is declared as an enum so the message the exception throws is not valid.

I see there is a test `testSplitWithEnumForZeroSizeClass` , is that what this check is for? (Nice job adding the test, I didn't realize that there could be empty enums).

If the check is for that then maybe the message should be changed? Just from reading the code I'd be tempted to remove the check as it appears as though it can't happen since E is always an enum.